Thanks for your pull request!

Ideally, I would like to get rid of the old behavior at all, since it is very inefficient.
However, reading a huge block of random data at once before writing it is also not very efficient memory wise.

Can you maybe come up with a middle way like reading and writing blocks of 64k?
At least 64k is also used when hashing files without mmap support: https://github.com/cfv-project/cfv/blob/79587c1ff05169ad67342e6fbc6a3c59dfec251a/lib/cfv/hash.py#L40

Please also fix the formatting error (see https://github.com/cfv-project/cfv/actions/runs/18862085679/job/54029609083?pr=84):
```
Run flake8 --exclude=build,venv --ignore= --max-line-length=200 --max-complexity=75 --show-source --statistics .
./test/benchmark.py:69:1: W293 blank line contains whitespace
    
^
1     W293 blank line contains whitespace
```

--- a/test/benchmark.py
+++ b/test/benchmark.py
@@ -70,10 +70,6 @@ def create_test_file(path, max_size, verbose=False, chunk_size=65536):
     size = random.randint(1, max_size)
     if verbose:
         print('creating', path, 'size', size)
-    
-    size = random.randint(1, max_size)
-    if verbose:
-        print("creating", path, "size", size, "chunk_size", chunk_size)
 
     written = 0
     try:

--- a/test/benchmark.py
+++ b/test/benchmark.py
@@ -62,15 +62,32 @@ def human_int(value):
     return int(value) * multiplier
 
 
-def create_test_file(path, max_size, verbose=False):
+def create_test_file(path, max_size, verbose=False, chunk_size=65535):
+    if max_size < 1:
+        raise ValueError("max_size must be >= 1")
+    if chunk_size < 1:
+        raise ValueError("chunk_size must be >= 1")
     size = random.randint(1, max_size)
     if verbose:
         print('creating', path, 'size', size)
-    with open(path, 'wb') as f:
-        # TODO: make this more efficient.
-        while size:
-            f.write(b'%c' % random.randint(0, 255))
-            size -= 1
+    
+    size = random.randint(1, max_size)
+    if verbose:
+        print("creating", path, "size", size, "chunk_size", chunk_size)
+
+    written = 0
+    try:
+        with open(path, "wb") as f:
+            remaining = size
+            while remaining > 0:
+                n = min(remaining, chunk_size)
+                f.write(os.urandom(n))
+                remaining -= n
+                written += n
+        return written
+    except OSError as e:
+        # Disk full, permission issues, etc. Propagate with context.
+        raise OSError(f"Failed to write {path} after {written} bytes: {e}") from e
 
 
 def create_test_dir(root, num_files, branch_factor, max_size, verbose=False):

--- a/test/benchmark.py
+++ b/test/benchmark.py
@@ -62,7 +62,7 @@ def human_int(value):
     return int(value) * multiplier
 
 
-def create_test_file(path, max_size, verbose=False, chunk_size=65535):
+def create_test_file(path, max_size, verbose=False, chunk_size=65536):
     if max_size < 1:
         raise ValueError("max_size must be >= 1")
     if chunk_size < 1:

